pkg/kubernetes/client.go

```diff
@@ -98,6 +102,21 @@ func (c *Client) initServicesMonitor() {
 	c.informers[Services].AddEventHandler(GetKubernetesEventHandlers((string)(Services), providerName, c.shouldObserve, svcEventTypes))
 }
 
+func (c *Client) initMultiClusterServicesMonitor() {
```

Comment:
Adding the changes here doesn't seem right, if I look at both the other custom crd's that we have in osm the changes for their informers are in these files : 

1. egress crd : https://github.com/openservicemesh/osm/blob/113bfc328b857272dc149066d162fc950f0b584f/pkg/policy/client.go#L41
2. meshConfig crd : https://github.com/openservicemesh/osm/blob/113bfc328b857272dc149066d162fc950f0b584f/pkg/configurator/client.go#L29

Please could we follow these examples and make the change for the mulitcluster crd accordingly

Reply:
I'd argue that those ways are more incorrect for the following reasons:

1. It places more direct dependencies on the MeshCatalog to Kubernetes (these should be abstracted through providers).
2. It's kind of messy to have a separate client for each CRD.. code will grow unwieldy
3. Knowledge of CRD's will become more intertwined, meaning more types will need to reconcile MultiClusterService with Service, or other CRD's. Forcing each type to need a client for each of these CRD's (when it's all reaching out to the Kuberentes master) is a burden on the developer. For a concrete example, we would then need to modify the kube provider to also have a MultiClusterServiceController.
4. Clients are clients that talk to a "thing". In this case that thing is the K8s master, why have 9 clients for 9 of the resources backed by that same "thing".

If you'd like, we can pick up the work to remove those 2 clients and migrate the code to the Kubernetes Controller.

Reply:
https://github.com/openservicemesh/osm/pull/3635 brings in those changes for the other client types